python/ffcc/io.py

```diff
@@ -14,145 +14,155 @@
+  if tf.io.gfile.exists(os.path.join(dir_path, scene_prefix + '_weight.txt')):
+    weight = np.loadtxt(os.path.join(dir_path, scene_prefix + '_weight.txt'),
+                        dtype=np.float32)
+  else:
+    weight = np.float32(1.0)
+  # load the extended feature if the weight file exists
+  if tf.io.gfile.exists(os.path.join(dir_path,
+                                     scene_prefix + '_extended_feature.txt')):
+    extended_feature = np.loadtxt(os.path.join(
+      dir_path, scene_prefix + '_extended_feature.txt'), dtype=np.float32)
+  else:
+    extended_feature = np.float32(1.0)
+  # load image and illuminant data
+  rgb = tf.io.decode_image(tf.io.read_file(
+    os.path.join(dir_path, scene_prefix + '.png')), dtype=tf.dtypes.uint16)
+  normalized = tf.cast(rgb, dtype=tf.float32) / tf.uint16.max
+  sess = tf.compat.v1.Session()
+  with sess:
+    rgb = sess.run(normalized)
   # Make sure the illuminant is unit vector.
   rgb_illuminant = np.loadtxt(
-      os.path.join(dir_path, 'illuminant.txt'), dtype=np.float32)
+    os.path.join(dir_path, scene_prefix + '.txt'), dtype=np.float32)
   rgb_illuminant /= np.linalg.norm(rgb_illuminant)
-  weight = np.loadtxt(os.path.join(dir_path, 'weight.txt'), dtype=np.float32)
+  return rgb_illuminant, weight, rgb, extended_feature
 
-  frames = []
-  for item in tf.io.gfile.listdir(dir_path):
-    sub_dir = os.path.join(dir_path, item)
-    if tf.io.gfile.isdir(sub_dir) and item.isdigit():
-      try:
-        rgb = sio.imread(os.path.join(sub_dir, 'image.tiff'))
-        extended_feature = np.loadtxt(
-            os.path.join(sub_dir, 'extended_feature.txt'), dtype=np.float32)
-        frames.append((rgb, extended_feature))
-      except:  # pylint: disable=bare-except
-        tf.logging.error('Unable to load %s', sub_dir)
 
-  return rgb_illuminant, weight, frames
-
-
-def encode_to_tfrecord(data, output_dir):
-  """Encodes a data item to TFRecord format.
+def get_training_eval_sets(files, cv_fold_index, test_fold):
+  """Retrieve train and evaluation sets for the current testing fold .
 
   Args:
-    data: A dictionary maps to the FFCC data bundle.
-    output_dir: The output folder. The record will be saved as <name>.tfrecord.
+    files: A list of filenames that should be synched with the cvfolds.
+    cv_fold_index: A list of corresponding fold number of each filename in
+      'files' for cross validation.
+    test_fold: Index of testing fold (should be in the range [1-3]).
+
+  Returns:
+    train_set and eval_set, lists of training and evaluation
+      filenames, respectively.
   """
+  assert len(files) == len(cv_fold_index)
+  assert test_fold in range(1, NUM_CROSS_VALIDATION_FOLDS + 1)
+  train_folds = list({1, 2, 3} - {test_fold})
+  train_set = []
+  for train_fold_i in train_folds:
+    train_set = list.append(train_set,
+                          np.array(files).transpose()[
+                            (cv_fold_index == train_fold_i)])
```

Comment:
Change to a regular python list comprehension? I dont't think we need numpy here.

Reply:
the problem I faced is that cv_fold_index is np array and this will give a logical array. Lists need either integer or slice. Any suggestions?

Reply:
Fixed!